### PR TITLE
simpler example for randn

### PR DESCRIPTION
**Pull request by lmiq**
As noted here: https://discourse.julialang.org/t/how-do-we-julians-win-big-when-the-situation-is-so-unfair/106433/63?u=lmiq

The randn example is overly complicated and does not indicate the simplest usage, which is probably the most typical one.

### CODE REVIEW

File: stdlib/Random/src/normal.jl [hunk 1]
```diff
@@ -23,8 +23,23 @@ from the circularly symmetric complex normal distribution of variance 1 (corresp
...
+
+```julia-repl
+julia> using Random
+
+julia> randn()
+-0.942481877315864
+```
+
+2. Setting up of the random number generator with a user-defined seed, and the
+generation of a random number with a `ComplexF32` type:
```
Comment (ViralBShah) [inline stdlib/Random/src/normal.jl:37]: The example below does `ComplexF64`.

**Comment by lmiq** [comment]
This is how it looks like in the current commit:

image

**Comment by stevengj** [comment]
I merged in a few more of my changes, hope you don't mind.

(I also noticed that the docs were out of date on the supported types due to #44714)

File: stdlib/Random/src/normal.jl [hunk 2]
```diff
@@ -14,25 +14,51 @@
...
+
+```julia-repl
+julia> randn(2,3)
+2×3 Matrix{Float64}:
+  1.18786   -0.678616   1.49463
+ -0.342792  -0.134299  -1.45005
+```
+
+Setting up of the random number generator `rng` with a user-defined seed (for reproducible numbers)
+and using it to generate a random `Float32` number or a matrix `ComplexF32` random numbers:
```
Comment (stevengj) [inline stdlib/Random/src/normal.jl:48]:
```suggestion
and using it to generate a random `Float32` number or a matrix of `ComplexF32` random numbers:
```

**Comment by PallHaraldsson** [comment]
Possibly you want to mention this is the "standard complex normal distribution", which I believe it is, i.e. same as from MATLAB:

https://www.mathworks.com/help/matlab/ref/randn.html#mw_6dadd167-e8bc-42bb-9aaa-9bcfb97f6815

I was expecting circularly symmetric (see that on stack overflow on a question for _uniform_, and also a square suggested, to show there isn't any one right, as opposed for randn I think).

Note, in MATLAB docs there's still an alternative to it. I.e. non-standard I would think. Is there a similar way to use randn?

All this got me thinking what are possible extensions to e.g. quaternions.

https://stackoverflow.com/questions/31600717/how-to-generate-a-random-quaternion-quickly

I'm not saying the docs should say anything on this, since Base doesn't have higher-complex, and such docs belong in relevant packages.

kieranwynn.github.io/pyquaternion/

Python doesn't have randn I think but numpy does, but it doesn't have for complex it seems, I see a closed and open issue: 
https://github.com/numpy/numpy/issues/13576

File: stdlib/Random/src/normal.jl [hunk 3]
```diff
@@ -14,25 +14,51 @@
 
 Generate a normally-distributed random number of type `T`
 with mean 0 and standard deviation 1.
-Optionally generate an array of normally-distributed random numbers.
-The `Base` module currently provides an implementation for the types
-[`Float16`](@ref), [`Float32`](@ref), and [`Float64`](@ref) (the default), and their
-[`Complex`](@ref) counterparts. When the type argument is complex, the values are drawn
-from the circularly symmetric complex normal distribution of variance 1 (corresponding to real and imaginary part having independent normal distribution with mean zero and variance `1/2`).
+Given the optional `dims` argument, generate an array of size `dims` of such numbers.
```
Comment (stevengj) [inline stdlib/Random/src/normal.jl:17]:
```suggestion
Given the optional `dims` argument(s), generate an array of size `dims` of such numbers.
```

**Pull request merged**